src/BaseUriPlugin.php

```diff
@@ -0,0 +1,72 @@
+<?php
+
+namespace Http\Client\Plugin;
+
+use Http\Promise\Promise;
+use Psr\Http\Message\RequestInterface;
+use Psr\Http\Message\UriInterface;
+
+class BaseUriPlugin implements Plugin
+{
+    /**
+     * @var UriInterface
+     */
+    private $baseUri;
+
+    /**
+     * @param UriInterface $baseUri
+     */
+    public function __construct(UriInterface $baseUri)
+    {
+        $this->baseUri = $baseUri;
+    }
+
+    /**
+     * Handle the request and return the response coming from the next callable.
+     *
+     * @param RequestInterface $request
+     * @param callable         $next    Next middleware in the chain, the request is passed as the first argument
+     * @param callable         $first   First middleware in the chain, used to to restart a request
+     *
+     * @return Promise
+     */
+    public function handleRequest(RequestInterface $request, callable $next, callable $first)
+    {
+        $uriString = $request->getUri()->__toString();
+        if (!strpos($uriString, '://')) {
```

Comment:
I think it's like @sagikazarmark said `//exemple.com/foo`

BTW guzzle and diactoros use parse_url as the parser for the string, so this string would recognize as the path : 

```
$ php -r "var_dump(parse_url('://exemple.com/test'));" 
array(1) {
  'path' =>
  string(19) "://exemple.com/test"
}

$ php -r "var_dump(parse_url('//exemple.com/test'));"
array(2) {
  'host' =>
  string(11) "exemple.com"
  'path' =>
  string(5) "/test"
}
```